rules/dead-code/tests/Rector/Class_/RemoveEmptyAbstractClassRector/Fixture/multi_extends.php.inc

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Rector\DeadCode\Tests\Rector\Class_\RemoveEmptyAbstractClassRector\Fixture;
+
+abstract class AnotherAbstractClassMulti
+{
+     public function getName()
+     {
+        return 'name';
+     }
+}
+
+abstract class SomeAbstractClassMulti1 extends AnotherAbstractClassMulti
+{
+}
+
+abstract class SomeAbstractClassMulti2 extends SomeAbstractClassMulti1
+{
+}
+
+class MultiExtends extends SomeAbstractClassMulti2
+{
+}
+
+?>
+-----
+<?php
+
+namespace Rector\DeadCode\Tests\Rector\Class_\RemoveEmptyAbstractClassRector\Fixture;
+
+abstract class AnotherAbstractClassMulti
+{
+     public function getName()
+     {
+        return 'name';
+     }
+}
+
+class MultiExtends extends AnotherAbstractClassMulti
+{
+}
+
+?>
```

Comment:
I miss a text fixture like this, where abstact class is actually used and skipped

Reply:
is the fixture incorrect?

Reply:
This is fixture is correct. But we'll need a new one that uses the abstract class and is skipped

Reply:
any example?

Reply:
E.g.


```php
final class SomeClass extends AbstractClass
{
}


final class AnyClass
{
	public function process($some)
    {
        if ($some instanceof AbstractClass) {
        }
    } 
}
```

If we revome `AbstractClass` here, the condition would break

Reply:
I added the fixture https://github.com/rectorphp/rector/pull/5558/commits/1e9fb8bdfde27050e7d14b6a5a6ecf109e1429c1 and it seems working ok 🎉 

![Screen Shot 2021-02-16 at 03 59 04](https://user-images.githubusercontent.com/459648/107992481-61a3ad80-700b-11eb-8f23-cf50835bd9fa.png)

Reply:
The removal is only for abstract class extends abstract class like in issue description.

Reply:
Ok, I got the failing fixture https://github.com/rectorphp/rector/pull/5558/commits/b40546f92e42dbb7819e5fdd7f3119730916895d

Reply:
@TomasVotruba is there a way to class the `FullyQualified` node used not in extends, like vendor locker?

Reply:
seems need to use BetterNodeFinder->find inside loop children

Reply:
@TomasVotruba I tried to collect `FullyQualified` node when class is not abstract but seems overlapped as abstract class detected first when abstract class in the top of non-abstract class.

Reply:
I've tried with `Instanceof_` to collect `FullyQualified` usage, and it overlapped as well, do you have idea to get it?

Reply:
@TomasVotruba using `FileNode` make it works 🎉

Reply:
> The removal is only for abstract class extends abstract class like in issue description.

Ah, I missed that.

> I've tried with Instanceof_ to collect FullyQualified usage, and it overlapped as well, do you have idea to get it?

This would have to be collected before running Rector rule, the same way `Class_` and other nodes are collected, see `RectorRepository`.

Reply:
I can't found class `RectorRepository`, is it has the service somewhere? or I need to create a new service `RectorRepository` with `FileNode` ?

Reply:
Let's merge it as it is to keep it one step at a time.

Please add a test fixture with 2 files in a next PR.

- first: one with class extending an abstract class + abstract class
- second: with abstract class usage

E.g. this test is using extra files technique: https://github.com/rectorphp/rector/blob/06906a06591c1c635669b269635352cfe4b36a2a/rules/autodiscovery/tests/Rector/FileNode/MoveInterfacesToContractNamespaceDirectoryRector/MoveInterfacesToContractNamespaceDirectoryRectorTest.php#L26

Reply:
> I can't found class RectorRepository, is it has the service somewhere? or I need to create a new service RectorRepository with FileNode ?

I'm sorry, it's `NodeRepository`

Reply:
Do we need new separate test class ? using `extraFiles` seems need manually added fixture files ?

Reply:
`NodeRepository`::collect()` seems not collect FullyQualified node.

Reply:
> Do we need new separate test class ? using extraFiles seems need manually added fixture files ?

Yes, that would be the best to keep it narrow :+1: 


> NodeRepository::collect()` seems not collect FullyQualified node.

It will have to be added. It should also collect simple `Name`, to cover cases like these:

```php
use App\Nested\AbstractClass;

public function useClass(-->AbstractClass<-- $abstractClass)
{
}
```

Reply:
I will try.